plugins/modules/idrac_diagnostics.py

```diff
@@ -0,0 +1,818 @@
+                if self.__validate_time(start_time):
+                    payload["ScheduledStartTime"] = start_time
+            if self.module.params.get('scheduled_end_time'):
+                end_time = self.__validate_time_format(self.module.params.get('scheduled_end_time'))
+                if self.__validate_time(end_time) and self.__validate_end_time(start_time, end_time):
+                    payload["UntilTime"] = end_time
+        payload["RebootJobType"] = reboot_job_types.get(reboot_type)
+        payload["RunMode"] = run_modes.get(run_mode)
+        run_diagnostics_status = self.idrac.invoke_request(self.run_url, "POST", data=payload)
+        return run_diagnostics_status
+
+    def __get_run_diagnostics_url(self):
+        uri, error_msg = validate_and_get_first_resource_id_uri(
+            self.module, self.idrac, MANAGERS_URI)
+        if error_msg:
+            self.module.exit_json(msg=error_msg, failed=True)
+        resp = get_dynamic_uri(self.idrac, uri)
+        url = resp.get('Links', {}).get(OEM, {}).get(MANUFACTURER, {}).get(LC_SERVICE, {}).get(ODATA, {})
+        if url:
+            action_resp = get_dynamic_uri(self.idrac, url)
+            run_url = action_resp.get(ACTIONS, {}).get(RUN, {}).get('target', {})
+            self.run_url = run_url
+        else:
+            self.module.exit_json(msg=UNSUPPORTED_FIRMWARE_MSG, failed=True)
+
+    def __validate_job_timeout(self):
+        if self.module.params.get("job_wait") and self.module.params.get("job_wait_timeout") <= 0:
+            self.module.exit_json(msg=TIMEOUT_NEGATIVE_OR_ZERO_MSG, failed=True)
+
+    def __perform_job_wait(self, run_diagnostics_status):
+        job_dict = {}
+        job_wait = self.module.params.get('job_wait')
+        job_wait_timeout = self.module.params.get('job_wait_timeout')
+        self.__validate_job_timeout()
+        job_tracking_uri = run_diagnostics_status.headers.get("Location")
+        if job_tracking_uri:
+            job_id = job_tracking_uri.split("/")[-1]
+            res_uri = validate_and_get_first_resource_id_uri(self.module, self.idrac, MANAGERS_URI)
+            job_uri = f"{res_uri[0]}/{OEM}/{MANUFACTURER}/{JOBS}/{job_id}"
+            if job_wait:
+                job_failed, msg, job_dict, wait_time = idrac_redfish_job_tracking(self.idrac, job_uri,
+                                                                                  max_job_wait_sec=job_wait_timeout,
+                                                                                  sleep_interval_secs=1)
+                job_dict = remove_key(job_dict, regex_pattern=ODATA_REGEX)
+                if int(wait_time) >= int(job_wait_timeout):
+                    self.module.exit_json(msg=WAIT_TIMEOUT_MSG.format(
+                        job_wait_timeout), changed=True, job_status=job_dict)
+                if job_failed:
+                    self.module.fail_json(
```

Comment:
lets use exit_json here to prevent traceback issue.

Reply:
done